src/onnx/onnx.cpp

```diff
@@ -470,7 +480,19 @@ struct onnx_parser
 
             if(s.find("SAME") != std::string::npos)
             {
-                op.padding_mode = op::padding_mode_t::same;
+                op.padding_mode                 = op::padding_mode_t::same;
+                std::vector<size_t> weight_dims = weights->get_shape().lens();
+                size_t weight_h                 = weight_dims[2];
+                size_t weight_w                 = weight_dims[3];
+
+                auto input_dims = l0->get_shape().lens();
+                std::vector<int64_t> padding(input_dims.size());
+                calculate_padding(
```

Comment:
we should also support same_lower by adding to flag to the function calculate_padding()

Reply:
Actually, you code change can be reused in both parse_conv and parse_pooling. I have  branch that calls your code change in this PR and used in both parse_conv and parse_pooling. The branch name is "correct_conv_shape_computation". This branch also support both the same_upper and same_lower mode. If you want, feel free to use the code changes in my branch.